### PR TITLE
Add Firefox support

### LINKED ISSUE #48: Firefox theme
If anyone else interested, I can take over and submit a PR to provide a Firefox theme.

### PR DESCRIPTION
**Pull request by andreicristianpetcu**
Fixes #48 

TODO: I was not able to generate the file locally. The "Download" button is not clickable and I don't know why :smile: 

Solarized dark.
Normal Window
image

Sidebar, popup and new tab text
image

Mega bar selection

image

Other themes

Solarized light 
image

Default dark
image

Default light

image

Polar ice

image

image

### CODE REVIEW

**Comment by kinduff** [comment]
Damn! This looks beautiful and works like a charm!

**Comment by andreicristianpetcu** [comment]
Firefox has more APIs available to themes. I used them :) @kinduff

**Comment by mjswensen** [comment]
Fantastic work, @andreicristianpetcu ! Really excellent!

I have a handful of housekeeping items to do on my end before this ships—let's change the target branch to `firefox` instead of `master` (since automatic deploys happen from `master), then I will get this cleaned up and fully merged as soon as possible!

Also, would you mind adding me to `themer-firefox` on the npm registry in case I need to push any patches?

Thank you so much again! All of our FF users will be so happy 😁

**Comment by mjswensen** [comment]
(Also, I'll fix up that issue with the file not downloading—really simple just missed a step in wiring everything up)

**Comment by andreicristianpetcu** [comment]
I can't change the target branch to firefox. Please create a firefox branch on your repo.
image

**Comment by andreicristianpetcu** [comment]
You are the author in npm. I'm fine with that.
image

**Comment by andreicristianpetcu** [comment]
Maybe create a persistent develop branch and I'll set that as the target. The firefox branch will be used once, a develop branch can be re-used by other PRs. @mjswensen

**Comment by mjswensen** [comment]
Great idea about the persistent branch. I've created and pushed a branch called `dev`—see if it will let you now do it now.

> You are the author on npm

Awesome, the only other thing you'll need to do is go to [the themer-firefox settings on npm](https://www.npmjs.com/package/themer-firefox/access) and add me as a maintainer. You are awesome!

**Comment by andreicristianpetcu** [comment]
I changed the dev branch. Will do the npm fix a bit later, not sure what my npm credentials are :)

**Comment by andreicristianpetcu** [comment]
That is not my account. I see @j4nr6n published the firefox-themer package. Should I rename mine?

**Comment by mjswensen** [comment]
Ahhh, that explains some confusion on my end! We will actually have two firefox packages (yes, I will rename yours to not clash on the name)—I will merge yours in as the officially supported theme, and reach out to the other author about potentially combining forces.

You are all set, no need to do anything on the npm side! I will take it from here and get this thing out as soon as I can. Will post back on this thread and on Twitter with updates.

Thank you again!!

File: cli/packages/themer-firefox/lib/index.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    hslColors: hslColorSet(colors[key]),
+    isDark: key === 'dark',
+  }));
+  return colorSets.map(colorSet => renderManifest(colorSet));
+};
+
+const renderInstructions = paths => {
+  const directories = new Set(paths.map(path.dirname));
+  return `
+1. npm i -g web-ext \`chrome://extensions\`.
```
Comment (j4nr6n) [inline cli/packages/themer-firefox/lib/index.js:79]: Isn't this extension for firefox?

**Comment by andreicristianpetcu** [comment]
uuups :) I copy/pasted from chrome

**Comment by j4nr6n** [comment]
I wasn't aware of `web-ext` at the time I created the extension, I want to revisit it now. It seems the manifest in this extension is more complete than the one from my extension as I just brought in the themed elements from the Chrome extension that have cross-browser support. I didn't submit a PR to this repository initially because I didn't see a request for Firefox support and I wasn't sure about the fact that you have to have an account to sign the extension before it can be installed in Firefox. I'd be happy to contribute some of the differences between the two projects and retire mine in favor of the officially supported one.

**Comment by andreicristianpetcu** [comment]
@j4nr6n if you want something cool for ff, take a look at this https://github.com/mjswensen/themer/issues/54

**Comment by mjswensen** [comment]
@j4nr6n , feel free to PR any improvements from your version into the `dev` branch to consolidate!

If you'd like, you can make me a maintainer on npm for the `themer-firefox` package (on the [settings page for the package](https://www.npmjs.com/package/themer-firefox/access), and I can release our consolidated version as an update. Or I can just release with a different name if you prefer, up to you.

Once the CLI package is ready to go, I'll finish up @andreicristianpetcu 's work on adding it to the web UI and we'll be live 🎉

**Comment by andreicristianpetcu** [comment]
I made a pr with the issue you noticed @j4nr6n https://github.com/mjswensen/themer/pull/55

**Pull request merged**